# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webcodecs/AudioData.cpp
# Commit: 874f792bb36f
# Full Hash: 874f792bb36f68f89c39862443d988748325ff88
# Author: Paul Adenot <paul@paul.cx>
# Date: 2024-03-06 04:41:40
# Regressor Bug: 1858958
# File Overlap Count: 1
# Description:
#   Bug 1858958 - Test more conversions, clearly mark planar->interleaved conversion as outside of spec. r=chunmin
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D192339
# ==============================================================================

diff -r 22ff6bc3cb95 -r 874f792bb36f dom/media/webcodecs/AudioData.cpp
--- a/dom/media/webcodecs/AudioData.cpp	Tue Mar 05 15:52:14 2024 +0000
+++ b/dom/media/webcodecs/AudioData.cpp	Tue Mar 05 15:52:15 2024 +0000
@@ -443,6 +443,7 @@
   }
 
   if (!IsInterleaved(aSourceFormat) && IsInterleaved(aCopyToSpec.mFormat)) {
+    MOZ_CRASH("This should never be hit -- current spec doesn't support it");
     // Planar to interleaved -- copy of all channels of the source into the
     // destination buffer.
     MOZ_ASSERT(aCopyToSpec.mPlaneIndex == 0);